### ISSUE TITLE
reward inconsistency with trl controlled sentiment generation example

### ISSUE THREAD

**Issue by EdoardoPona** [EdoardoPona/predicting-inductive-biases-RL#25]
The trl sentiment generation example uses a reward that is slightly different from ours.
Given @diogo-cruz 's experiments, showing that the modified trl example converges better than our rl4lm code, raises the question whether (besides trl optimisations for VRAM) it is this difference in reward that is responsible. 
It is worth trying our rl4lm setup with a reward similar to the one in the trl example. 

**trl reward** 
after passing through a sentiment classification pipeline, obtaining logits for 3 (negative, neutral, positive) classes, the reward is implemented as such 
```python
def pos_logit_to_reward(logit, task):
    """
    Take the positive sentiment logit and scale it for the task.
        task [negative]: reward = -logit
        task [neutral]: reward = -2*abs(logit)+4
        task [positive]: reward = logit
    """
```
key thing here is that the reward for the negative sentiment is ` -logit[positive]` rather than `logit[negative]` which is the case in our rl4lm implementation. I don't quite understand / agree with this choice over ours, but it is definitely worth trying 

**task**
implement a similar reward for our rl4lm setup and test it.

**Comment by EdoardoPona** [comment]
implemented in https://github.com/diogo-cruz/RL4LMs/commit/d105bdc99c4bc9e4b294ec2b105fc9791a4b88c6

to be tested on a run

**Comment by EdoardoPona** [comment]
bearing in mind that we are measuring reward for the final rate/reward plots, does this type of reward change anything?
